apps/core/src/hooks/useTransactionSummary.ts

```diff
@@ -45,6 +50,140 @@ const getSummary = (
     }
 };
 
+const getObjectSummaryNFTData = <T extends { objectId: string }>({
+    nftsMeta,
+    originalData,
+}: {
+    nftsMeta: ReturnType<typeof useNFTsMeta>;
+    originalData: T[];
+}) => {
+    const objectIdsFromNFTData = nftsMeta.data.objectIds;
+    const objectIdsFromNFTDataSet = new Set(objectIdsFromNFTData);
+
+    return originalData.reduce((acc, obj) => {
+        if (objectIdsFromNFTDataSet.has(obj.objectId)) {
+            const index = nftsMeta.data.objectIds.indexOf(obj.objectId);
+
+            return [
+                ...acc,
+                {
+                    ...obj,
+                    nftMeta: nftsMeta.data.metaData[index],
```

Comment:
I'd avoid this and instead just have  a property that's the original property, then separately have a `nftMeta` property